acai-ios/MenuItem.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  CustomizationOption.swift
+//  acai-ios
+//
+//  Created by Artesia Ko on 3/5/19.
+//  Copyright © 2019 Cornell AppDev. All rights reserved.
+//
+
+import Foundation
+import IGListKit
+
+class MenuItem: ListDiffable {
+    
+    var title: String = ""
+    var basePrice: Double = 0
+    var image: UIImage = UIImage()
+    var customizationOptions: [OrderCustomizationOption] = []
+    
+    init (title: String, basePrice: Double, customizationOptions: [OrderCustomizationOption], image: UIImage) {
+        self.title = title
+        self.basePrice = basePrice
+        self.customizationOptions = customizationOptions
+        self.image = image
+    }
+    
+    func getSelectedSubOptionsPrice() -> Double {
+        var selectedSubOptionsPrice: Double = 0
+        for option in customizationOptions {
+            for subOption in option.options {
+                if (subOption.isSelected) {
+                    selectedSubOptionsPrice += subOption.price
+                }
+            }
+        }
+        return selectedSubOptionsPrice
+    } 
+    
+    func getSelectedSubOptionsText() -> String {
+        var selectedSubOptionsText: String = ""
+        for option in customizationOptions {
+            for subOption in option.options {
+                if (subOption.isSelected) {
+                    if selectedSubOptionsText == "" {
+                            selectedSubOptionsText = "\(subOption.title)"
+                    }
+                    else {
+                        selectedSubOptionsText = "\(selectedSubOptionsText), \(subOption.title)"
+                    }
+                }
+            }
+        }
+        return selectedSubOptionsText
+    }
+    
+    func getSelectedSubOptions() -> [OrderCustomizationSubOption] {
+        var selectedSubOptions: [OrderCustomizationSubOption] = []
+        for option in customizationOptions {
+            for subOption in option.options {
+                if (subOption.isSelected) {
+                    selectedSubOptions.append(subOption)
+                }
+            }
+        }
```

Comment:
here you can use a `forEach` for the outer loop and a `filter` for the inner loop